felix/fv/qos_controls_test.go

```diff
@@ -156,6 +164,106 @@ var _ = infrastructure.DatastoreDescribe(
 
 			})
 		})
+
+		Context("With packet rate limits", func() {
+			getRules := func(felixId int) string {
+				out, err := tc.Felixes[felixId].ExecOutput("iptables-save", "-c")
+				log.Infof("iptables-save -c output:\n%v", out)
+				Expect(err).NotTo(HaveOccurred())
+				return out
+			}
+
+			It("should limit packet rate correctly", func() {
+				format.MaxLength = 1000000
+
+				By("Starting iperf3 server on workload 0")
+				serverCmd := w[0].ExecCommand("iperf3", "-s")
+				err := serverCmd.Start()
+				Expect(err).NotTo(HaveOccurred())
+
+				By("Running iperf3 client on workload 1")
+				out, err := w[1].ExecOutput("iperf3", "-c", w[0].IP, "-O5", "-M1000", "-J")
+				Expect(err).NotTo(HaveOccurred())
+				baselineRate, err := getRateFromJsonOutput(out)
+				Expect(err).NotTo(HaveOccurred())
+				log.Infof("iperf client rate with no packet rate limit (bps): %v", baselineRate)
+				// Expect the baseline rate to be much greater (>=100x) the bandwidth that we
+				// would get with the packet rate we are going to configure just below. In
+				// practice we see several Gbps here.
+				Expect(baselineRate).To(BeNumerically(">=", 800000.0*100))
```

Comment:
Is there a connection between 800000 here and `IngressPacketRate: 100` below?

Reply:
Not exactly a connection, just expecting 100x+ faster traffic with no limits than when limited. I'll add a comment for how we arrived at 800000 here too (as below)

Reply:
The `*100` part is fine.  It's the `800000` that needs explanation.  I know now that this is explained a bit further below - but I tend to read code forwards!